File: apps/atlas-api/src/features/home/presentation/http/handle-probe-metrics.ts
import type { Telemetry } from "../../../telemetry/application/ports/telemetry";

/** Flag-gated enhanced probe — emits home-probe OTLP counters. */
export function handleProbeMetrics(telemetry: Telemetry): Response {
  // @fireweave-controlpoint home-probe-metrics
  try {
    telemetry.increment("feature.home-probe.adopted");
    return Response.json({
      ok: true,
      feature: "home-probe-metrics",
      emitted: ["feature.home-probe.adopted"],
    });
  } catch (err) {
    telemetry.increment("feature.home-probe.error");
    return Response.json(
      {
        ok: false,
        error: err instanceof Error ? err.message : String(err),
      },
      { status: 500 },
    );
  }
}

File: apps/lantern-api/src/features/home/presentation/http/handle_probe_metrics.py
import json

from src.features.telemetry.infrastructure.start_otel import Telemetry


def handle_probe_metrics(telemetry: Telemetry) -> tuple[int, bytes]:
    # @fireweave-controlpoint home-probe-metrics
    try:
        telemetry.increment("feature.home-probe.adopted")
        return 200, (
            json.dumps(
                {
                    "ok": True,
                    "feature": "home-probe-metrics",
                    "emitted": ["feature.home-probe.adopted"],
                }
            )
            + "\n"
        ).encode()
    except Exception as err:
        telemetry.increment("feature.home-probe.error")
        return 500, (json.dumps({"ok": False, "error": str(err)}) + "\n").encode()
